### PR TITLE
flip tag and status options fields; add stub data; add option to fulfill and notify;

### PR DESCRIPTION
**Pull request by tekhaus**
Since this task runs on order update, I decided to just rely on the fulfillment (if option chosen) to cause another update and thus hitting the other branch that checks tags on the next task run, in lieu of getting overly complex with actions/perform or a custom event.

I purposely only try to fulfill order with NO prior fulfillments, so if an order is manually fulfilled and cancelled, and then tagged, it will not try to fulfill it again.

I wasn't sure whether it was useful to add an additional preview block so the fulfillment would evaluate. Let me know if I should.

Also, in the slack you advised against relying on the **task.shopify_api_version** values, but with these versioned REST calls this actually seems like a perfect use for them.

### CODE REVIEW

**Comment by tekhaus** [comment]
[Modernize "Manage fulfillment status using order tags"](https://trello.com/c/NdKovFiS/15-modernize-manage-fulfillment-status-using-order-tags)

File: docs/manage-fulfillment-status-using-order-tags/script.liquid [hunk 1]
```diff
@@ -1,45 +1,78 @@
...
-        {"log": "Fulfillment status is already {{ desired_status }}; nothing to do"}
+        {% log %}
+          "Fulfillment status is already {{ desired_status }}; nothing to do"
+        {% endlog %}
       {% else %}
         {% action "shopify" %}
           [
             "post",
-            "/admin/api/2019-04/orders/{{ order.id }}/fulfillments/{{ fulfillment.id }}/events.json",
+            "/admin/api/{{ task.shopify_api_version }}/orders/{{ order.id }}/fulfillments/{{ fulfillment.id }}/events.json",
```
Comment (isaacbowen) [inline docs/manage-fulfillment-status-using-order-tags/script.liquid:75]: Huh, that _is_ clever. :D Nice.

File: docs/manage-fulfillment-status-using-order-tags/script.liquid [hunk 2]
```diff
@@ -1,45 +1,78 @@
...
+          }
+        ]
+      {% endaction %}
+
+      {% break %}
+    {% endif %}
+  {% endfor %}
+
+  {% log "No shipment status tags found on this order; skipping" %}
+{% else%}
```
Comment (isaacbowen) [inline docs/manage-fulfillment-status-using-order-tags/script.liquid:42]: While you're in here for the documentation update (if you don't mind!), can you add a space on the other end of this `else`?

**Review by isaacbowen — CHANGES_REQUESTED** [review]
> Since this task runs on order update, I decided to just rely on the fulfillment (if option chosen) to cause another update and thus hitting the other branch that checks tags on the next task run, in lieu of getting overly complex with actions/perform or a custom event.

This was an excellent opportunity to choose simplicity. Bravo. 👏 

> I purposely only try to fulfill order with NO prior fulfillments, so if an order is manually fulfilled and cancelled, and then tagged, it will not try to fulfill it again.

This is a good thing to include in the task documentation. Can you add that in? That (and the whitespace nit) is my only requested change - this is a _great_ contribution. 🙌 

> I wasn't sure whether it was useful to add an additional preview block so the fulfillment would evaluate. Let me know if I should.

I am also not sure. :D I'm okay with what you've chosen.

**Pull request merged**